crates/turborepo-lib/src/package_manager/mod.rs

```diff
@@ -241,7 +253,36 @@ impl PackageManager {
     pub fn get_workspace_globs(
         &self,
         root_path: &AbsoluteSystemPath,
-    ) -> std::result::Result<Globs, Error> {
+    ) -> Result<WorkspaceGlobs, Error> {
+        let (mut inclusions, mut exclusions) = self.get_configured_workspace_globs(root_path)?;
+        exclusions.extend(self.get_default_exclusions().into_iter());
+
+        // Yarn appends node_modules to every other glob specified
+        if *self == PackageManager::Yarn {
+            inclusions
+                .iter_mut()
+                .for_each(|inclusion| exclusions.push(inclusion.clone() + "/node_modules/**"));
+        }
+        let globs = WorkspaceGlobs::new(inclusions, exclusions)?;
+        Ok(globs)
+    }
+
+    fn get_default_exclusions(&self) -> Vec<String> {
+        let ignores = match self {
+            PackageManager::Pnpm | PackageManager::Pnpm6 => {
+                vec!["**/node_modules/**", "**/bower_components/**"]
+            }
+            PackageManager::Npm => vec!["**/node_modules/**"],
+            PackageManager::Berry => vec!["**/node_modules", "**/.git", "**/.yarn"],
+            PackageManager::Yarn => vec![], // yarn does its own handling above
+        };
+        ignores.into_iter().map(|s| s.to_string()).collect()
+    }
```

Comment:
Since we're just using this to extend we can avoid allocating two vecs and instead return an iterator
```suggestion
    fn get_default_exclusions(&self) -> impl Iterator<Item = String> {
        let ignores = match self {
            PackageManager::Pnpm | PackageManager::Pnpm6 => {
                ["**/node_modules/**", "**/bower_components/**"].as_slice()
            }
            PackageManager::Npm => ["**/node_modules/**"].as_slice(),
            PackageManager::Berry => ["**/node_modules", "**/.git", "**/.yarn"].as_slice(),
            PackageManager::Yarn => [].as_slice(), // yarn does its own handling above
        };
        ignores.iter().map(|s| s.to_string())
    }
```